### PR TITLE
fix : wrong dates displayed in ScheduleFragment

### LINKED ISSUE #1541: Wrong dates displayed in DaySchedularFragment.
**Actual Behaviour**
If the user is using the app with the following configuration
screenshot_20170510-054441
then dates displayed in the `DaySchedularFragment` are incorrect.

eg when you open the sessions under the `16Jan` tab, the date displayed in the `SessionDetailActivity` for that session is 17 Jan.

**Expected Behaviour**
The dates in `DaySchedularFragment` and `SessionDetailActivity` should be same.

**Would you like to work on the issue?**
No

**Target Milestone to solve the issue**

Till which [milestone](https://github.com/fossasia/open-event-android/milestones) will you be fixing the issue?

### PR DESCRIPTION
**Pull request by Shailesh351**
Fixes #1541 

Changes:

- Changed 'D' to 'd' because 'D' is for 'Day in year' & 'd' is for 'Day in month'  
- set event time zone on simple date formate
 
Screenshots for the change: 
fix-date

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/utils/ISO8601Date.java [hunk 1]
```diff
@@ -51,12 +51,12 @@ public static String dateFromCalendar(Calendar currentDate) {
...
 
     public static String getTimeZoneDateFromString(String dateString) throws ParseException {
-        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-DD", Locale.getDefault());
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
         formatter.setTimeZone(getEventTimezone());
         Date date = formatter.parse(dateString);
-        SimpleDateFormat newFormat = new SimpleDateFormat("d MMM", Locale.getDefault());
-
-        return newFormat.format(date);
+        SimpleDateFormat newFormat = new SimpleDateFormat("d MMM, z", Locale.getDefault());
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/ISO8601Date.java]: Is `z` needed?

It is again split out below, so...

Comment (Shailesh351) [inline android/app/src/main/java/org/fossasia/openevent/utils/ISO8601Date.java]: Yes i have checked it on my device. Without 'z' it is formatting date with "GMT+5:30" for both time zones that's why i am using 'z'

Just check it using Log in getTimeZoneDateFromString() function
`dateString +"  "+ date.toString()`

Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/utils/ISO8601Date.java]: Ok. For an event based application, these date errors are crucial. I would have recommended Joda Time library but it's size is huge relative to the convenience it provides. What we can do is write tests. Currently, I am busy with exams so probably you or some other person can take up the issue to create unit tests for date conversion code?

**Comment by Shailesh351** [comment]
Thanks @iamareebjamal @arp95

**Pull request merged**